src/arch/x86/analyzex86.rs

```diff
@@ -504,61 +480,28 @@ fn binary_read(
     address: usize,  
     value_size: usize,
     base_addr: usize,
-    binary: &mut [u8])->i64
+    binary: &mut [u8]) -> i64
 {
-    match value_size 
+    match value_size
     {
-        1=>{
-            let mut temp: u8 = 0;
-            if read_int::<u8>(
-                &mut temp, 
-                address as usize, 
-                base_addr,
-                binary.len(),
-                binary)
-            {
-                return (temp as i8) as i64;
-            }
+        1 => {
+            read_int::<u8>(address, base_addr, binary)
```

Comment:
oop, read_int can read signed values now so I should make this `<i8>` and remove the `as i8` afterward